lib/backend/memory/memory.go

```diff
@@ -274,7 +275,7 @@ func (m *Memory) Delete(ctx context.Context, key []byte) error {
 	m.Lock()
 	defer m.Unlock()
 	m.removeExpired()
-	if m.tree.Get(&btreeItem{Item: backend.Item{Key: key}}) == nil {
+	if !m.tree.Has(&btreeItem{Item: backend.Item{Key: key}}) {
```

Comment:
Do we need `breeItem` at all anymore? It would be nice if we didn't have to allocate a `btreeItem` just to do these `Has` checks.